src/poll.rs

```diff
@@ -188,6 +188,7 @@ impl ZmqPoller {
     fn clear_read_ready(&self, cx: &mut Context<'_>) -> Result<()> {
         if let Poll::Ready(mut guard) = self.0.poll_read_ready(cx)? {
             guard.clear_ready();
```

Comment:
`poll_read_ready` stores `Waker` so that the task will be waken up on ready only if it returns `Pending`.
`clear_ready` in `tokio 1.0` only clears readiness. So, nobody setup `Waker` at this point. If this task just returns `Pending` here, it will never be polled again.

So, `cx.waker().wake_by_ref()` seems required so that the task is polled again (as the corresponding logic in `tokio 0.2` does).

According to my local experiment, the issue happens only when it's called inside `tokio::spawn`. It doesn't happen inside `block_on` (=directly under `tokio::test/main`). In `block_on`, the task is polled multiple times for some reason even after returning `Pending` without storing `Waker`, then `poll_read_ready`  setup `Waker` in the next poll. On the other hand, if the task is in `tokio::spawn`, it is polled exactly once when necessary. So without this patch, it ends up `Pending` without `Waker` setup, and never be polled again.